airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/manifest.yaml

```diff
@@ -0,0 +1,2993 @@
+              {% endif %}
+            {% endfor %}
+
+            {% if not ns.cursor %}
+              - startDate: "{{ config['date_ranges_start_date'] }}"
+                endDate: "{{ format_datetime(config['date_ranges_end_date'] if config.get('date_ranges_end_date') else now_utc(), '%Y-%m-%d') }}"
+            {% else %}
+              - startDate: "{{ '{{ stream_slice.start_time }}' }}"
+                endDate: "{{ '{{ stream_slice.end_time }}' }}"
+            {% endif %}
+        # This mapping adds the incremental_sync component for streams that support incremental loads.
+        # It is enabled if any of the following fields are present in the dimensions list:
+        # "date", "yearWeek", "yearMonth", or "year"
+        - type: ComponentMappingDefinition
+          field_path: ["incremental_sync"]
+          create_or_update: True
+          value: >
+            {% set ns = namespace(cursor="") %}
+            {% set cohort = components_values.get("cohortSpec") %}
+            {% set dimensions = components_values["dimensions"] %}
+            {% set date_fields = ["date", "yearWeek", "yearMonth", "year"] %}
+
+            {% for field in date_fields %}
+              {% if not ns.cursor and field in dimensions %}
+                {% set ns.cursor = field %}
+              {% endif %}
+            {% endfor %}
+
+            {% if ns.cursor and not (cohort and cohort.get("enabled")) %}
+              type: DatetimeBasedCursor
+              cursor_field: {{ ns.cursor }}
+              lookback_window: "P{{ config.get('lookback_window', 0) }}D"
+              step: "P{{ config.get('window_in_days', 365) }}D"
+              cursor_granularity: PT1S
+              cursor_datetime_formats:
+                - "%Y%m%d"
+              datetime_format: "%Y-%m-%d"
+              start_datetime:
+                type: MinMaxDatetime
+                datetime: "{% raw %}{{ config['date_ranges_start_date'] }}{% endraw %}"
+                datetime_format: "%Y-%m-%d"
+              end_datetime:
+                type: MinMaxDatetime
+                datetime: "{% raw %}{{ format_datetime(config['date_ranges_end_date'] if config.get('date_ranges_end_date') else now_utc(), '%Y-%m-%d') }}{% endraw %}"
+                datetime_format: "%Y-%m-%d"
+            {% endif %}
+        # This is a pretty tricky part where we have two levels of interpolation.
+        # First, we resolve components_values["metrics"] at the stream-template parsing level.
+        # Then, after creating the stream instance, we resolve and send information using the record,
+        # where the record represents the final schema field definition
+        # (e.g., {<field_name>: [None, <data_type>]} for schema filtering).
+        - type: ComponentMappingDefinition
+          field_path: ["schema_loader", "schema_filter", "condition"]
+          value: >
+            {{ '{{ record[0] in ' ~ (components_values["metrics"] | tojson) ~ ' }}' }}
```

Comment:
I think the above comment is good, but can you mention that it is also designed so we don't try to interpolate `records` because it doesn't exist when we build the template, but that will be part of the filter expression during the sync

Reply:
Comment in the code updated